Implemented new check ``unnecessary-dict-index-lookup`` (#4485)
PyCQA/pylint: doc/whatsnew/2.9.rst

* An ``ignore_signatures`` option has been added to the similarity checker. It will permits  to reduce false positives when multiple functions have the same parameters.

> * ``unnecessary-dict-index-lookup``: Emitted when iterating over dictionary items (key-value pairs) and
>   then indexing the same dictionary with the key within loop body.

Reviewer: ```suggestion
* ``unnecessary-dict-index-lookup``: Emitted when iterating over dictionary items
  (key-value pairs) and  accessing the value by index lookup.
```

Author: fixed 0b414d3a
